lib/src/Base/Stat/openturns/KroneckerCovarianceModel.hxx

```diff
@@ -0,0 +1,127 @@
+                          const Point &amplitude);
+
+  /** Standard constructor with scale and amplitude parameters parameters */
+  KroneckerCovarianceModel(const CovarianceModel & rho,
+                          const Point & amplitude,
+                          const CorrelationMatrix & spatialCorrelation);
+
+  /** Standard constructor with scale, amplitude and spatial correlation parameters */
+  KroneckerCovarianceModel(const CovarianceModel &rho,
+                          const CovarianceMatrix &sigma);
+
+  /** Virtual copy constructor */
+  KroneckerCovarianceModel * clone() const override;
+
+  /** Computation of the covariance function */
+  using CovarianceModelImplementation::operator();
+  SquareMatrix operator()(const Point &s, const Point &t) const override;
+  SquareMatrix operator()(const Point &tau) const override;
+
+  /** Computation of the covariance function */
+  using CovarianceModelImplementation::computeAsScalar;
+  Scalar computeAsScalar(const Point &s,
+                         const Point &t) const override;
+  Scalar computeAsScalar(const Point &tau) const override;
+#ifndef SWIG
+  Scalar computeAsScalar(const Collection<Scalar>::const_iterator &s_begin,
+                         const Collection<Scalar>::const_iterator &t_begin) const override;
+#endif
+
+  /** Discretize the covariance function on a given TimeGrid */
+  using CovarianceModelImplementation::discretize;
+  CovarianceMatrix discretize(const Sample & vertices) const override;
+  using CovarianceModelImplementation::discretizeAndFactorize;
+  TriangularMatrix discretizeAndFactorize(const Sample &vertices) const override;
+
+  /** Gradient */
+  using CovarianceModelImplementation::partialGradient;
+  Matrix partialGradient(const Point &s, const Point &t) const override;
+
+  /** isParallel? */
+  Bool isParallel() const override;
+
+  /** String converter */
+  String __repr__() const override;
+
+  /** String converter */
+  String __str__(const String & offset = "") const override;
+
+  /** Parameter accessor */
+  Point getFullParameter() const override;
+  void setFullParameter(const Point & parameter) override;
+  Description getFullParameterDescription() const override;
+  void setScale(const Point &scale) override;
+
+  /** Method save() stores the object through the StorageManager */
+  void save(Advocate & adv) const override;
+
+  /** Method load() reloads the object from the StorageManager */
+  void load(Advocate & adv) override;
+
+protected:
+  friend struct KroneckerModelDiscretizePolicy;
+  void setCorrelationModel(const CovarianceModel & rho);
+  Scalar computeRhoOverSample(const Collection<Scalar>::const_iterator &s_begin,
+                              const Collection<Scalar>::const_iterator &t_begin) const;
+
+private :
+  
+  // kernel (or correlation) scalar model
+  CovarianceModel rho_;
+
+  // Number of extra parameter (not scale/amplitude)
+  UnsignedInteger extraParameter_;
```

Comment:
in ProductCovarianceModel we have extraParameterNumber_

Reply:
indeed we might have similar variables